fv3gfs/util/buffer.py

```diff
@@ -1,17 +1,73 @@
+from __future__ import annotations
 from typing import Callable, Iterable, Optional, Dict, Tuple
 from ._timing import Timer, NullTimer
-from numpy import ndarray
+import numpy as np
 import contextlib
-
-from .utils import assign_array, is_c_contiguous
+from .utils import is_c_contiguous
 from .types import Allocator
 
+BufferKey_t = Tuple[Callable, Iterable[int], type]
```

Comment:
@mcgibbon : an opinion on this "typedef" ? Not very pythony I assume, but I'd rather not duplicate the Tuple[...] below. I Used a common C pattern for the naming, also happy to change that

Reply:
It's great! The only change I'd suggest is just call it `BufferKey`, which is normal for custom typedefs like this.

Reply:
This should also probably be imported into `fv3gfs.util.__init__.py`.